test/src/test/java/hudson/model/FileParameterValueTest.java

```diff
@@ -325,6 +325,28 @@ public void fileParameter_canStillUse_internalHierarchy() throws Exception {
         assertThat(workspaceParentContent, containsString("child2.txt"));
     }
 
+    @Test
+    @Issue("JENKINS-12999")
+    public void fileParameter_deleteTmpFile() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+            p.addProperty(new ParametersDefinitionProperty(Collections.singletonList(
+                    new FileParameterDefinition("fpv_test", null)
+        )));
+            
+        String uploadedContent = "test-content";
+        File uploadedFile = tmp.newFile();
+        FileUtils.write(uploadedFile, uploadedContent);
+            
+    	  FileParameterValue fpv = 
+    		    new FileParameterValue("fpv_test", uploadedFile, "uploaded-file.txt");
+        FreeStyleBuild build = 
+    		    p.scheduleBuild2(0, new Cause.UserIdCause(), new ParametersAction(fpv)).get();
+            
+    	  assertThat(build.getResult(), equalTo(Result.SUCCESS));
+        // confirm that the tmp file has been deleted
+    	  assertThat(fpv.getFile(), equalTo(null));
```

Comment:
assertNull() probably easier to read (I'm not doing the suggestion directly because it might require the import).